runtime/compiler/control/JITaaSCompilationThread.cpp

```diff
@@ -313,7 +361,62 @@ bool handleServerMessage(JITaaS::J9ClientStream *client, TR_J9VM *fe)
          TR_ResolvedMethod *method2 = std::get<2>(recv);
          int32_t cpIndex2 = std::get<3>(recv);
          int32_t isStatic = std::get<4>(recv);
-         client->write(fe->jitFieldsAreSame(method1, cpIndex1, method2, cpIndex2, isStatic));
+         bool identical = false;
+         bool compareFields = false;
+         UDATA f1 = 0, f2 = 0;
+         J9Class *declaringClass1 = NULL, *declaringClass2 = NULL;
+         J9ConstantPool *cp1 = (J9ConstantPool *) method1->ramConstantPool();
+         J9ConstantPool *cp2 = (J9ConstantPool *) method2->ramConstantPool();
+
+         J9RAMFieldRef *ramRef1 = (J9RAMFieldRef*) &(((J9RAMConstantPoolItem *)cp1)[cpIndex1]);
```

Comment:
Please add a comment with information where this code comes from. If the VM code ever changes and our caching validation catches the error, we need to remember where to look for possible discrepancies.